modules/dag/src/main/scala/org/tesselation/dag/l1/DAGBlock.scala

```diff
@@ -0,0 +1,21 @@
+package org.tesselation.dag.l1
+
+import cats.data.NonEmptyList
+
+import org.tesselation.schema.L1Block
+import org.tesselation.schema.height.Height
+import org.tesselation.schema.transaction.Transaction
+import org.tesselation.security.hash.Hash
+import org.tesselation.security.signature.Signed
+
+import io.estatico.newtype.ops.toCoercibleIdOps
+
+case class BlockReference(hash: Hash, height: Height)
+
+case class DAGBlock(transactions: Seq[Signed[Transaction]], parent: NonEmptyList[BlockReference]) extends L1Block {
+
+  // TODO: looks ugly
+  // TODO: confirm if we should increment the lower or higher parent height value
```

Comment:
Probably need someone to double check that this is still using same definition as earlier -- but height should be a max. It's supposed to represent how many references are required to trace back to some origin. With 2 parents of height 10 and 20, you'd need to trace back a maximum of (20+1)=21 references for the new child

Reply:
Currently we're using the maxheight for doing diffs before redownload. We're using minheight for pretty much everything else. We're don't need to worry about greatest common ancestors except for optimizing tip conflict handling in the L0.